test/Grpc.IntegrationTests/OrchestrationPatterns.cs

```diff
@@ -561,8 +561,37 @@ public async Task SpecialSerialization()
         Assert.Equal("new value", output?["newProperty"]?.ToString());
     }
 
+    // TODO: Additional versioning tests
+    [Fact]
+    public async Task OrchestrationVersionPassedThroughContext()
+    {
+        var version = "0.1";
+        await using HostTestLifetime server = await this.StartWorkerAsync(b =>
+        {
+            b.AddTasks(tasks => tasks
+                .AddOrchestratorFunc<string, string>("Versioned_Orchestration", (ctx, input) =>
+                {
+                    return ctx.CallActivityAsync<string>("Versioned_Activity", ctx.Version);
+                })
+                .AddActivityFunc<string, string>("Versioned_Activity", (ctx, input) =>
+                {
+                    return $"Orchestration version: {input}";
+                }));
+        }, c =>
+        {
+            c.UseDefaultVersion(version);
+        });
+
+        var instanceId = await server.Client.ScheduleNewOrchestrationInstanceAsync("Versioned_Orchestration", input: string.Empty);
+        var result = await server.Client.WaitForInstanceCompletionAsync(instanceId, getInputsAndOutputs: true, this.TimeoutToken);
```

Comment:
I'm realizing just now that we don't have a `Version` property in `OrchestrationMetadata`. We should add this, but it doesn't have to be part of this PR since it would also likely require backend changes.

Reply:
Yeah, I can follow up with this, what I'd like to add next will involve backend changes eventually so I'll be there soon, hopefully.